src/components/Restaurant.tsx

```diff
@@ -0,0 +1,80 @@
+import { Component } from 'react';
+import { RestaurantState } from '../types/component';
+import { saveToLocalStorage } from '../utils/localStorage';
+import { getRestaurantListData } from '../data/restaurantListData';
+import { filterAndSortRestaurantList } from '../domains/restaurantUtil';
+import FilterSection from './FilterSection';
+import RestaurantList from './RestaurantList';
+import Modal from './Modal';
+import RestaurantDetail from './RestaurantDetail';
+
+class Restaurant extends Component {
+  state: RestaurantState;
+
+  constructor(props: {}) {
+    super(props);
+
+    const restaurantList = filterAndSortRestaurantList(getRestaurantListData());
+
+    this.state = {
+      restaurantList: restaurantList,
+      currentRestaurantList: restaurantList,
+      selectedRestaurant: null,
+      isModalOpen: false,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('beforeunload', () => {
+      saveToLocalStorage(this.state.restaurantList);
+    });
+  }
+
+  updateCurrentRestaurantList = (filter: string, sortBy: string) => {
+    const updatedRestaurantList = filterAndSortRestaurantList(
+      this.state.restaurantList,
+      filter,
+      sortBy
+    );
+    this.setState({ currentRestaurantList: updatedRestaurantList });
+  };
+
+  updateSelectedRestaurant = (id: number) => {
+    const selectedRestaurant = this.state.currentRestaurantList.find(
+      (restaurant) => restaurant.id === id
+    );
+
+    if (!selectedRestaurant) return;
+
+    this.setState({ selectedRestaurant });
+    this.toggleIsModalOpen();
+  };
+
+  toggleIsModalOpen = () => {
+    this.setState({ isModalOpen: !this.state.isModalOpen });
```

Comment:
```suggestion
    this.setState((prev) => ({ ...prev, isModalOpen: !prev.isModalOpen }));
```

요렇게 함수형으로도 쓸 수 있어요~

(타입에러가 날텐데 요건 숙제로 ..)